src/zimscraperlib/zim/dedup.py

```diff
@@ -0,0 +1,73 @@
+import pathlib
+import re
+from typing import Any
+
+import xxhash
+
+from zimscraperlib.zim.creator import Creator
+
+
+class Deduplicator:
+    """Automatically deduplicate potential ZIM items before adding them to the ZIM
+
+    This class automatically computes the digest of every item added to the ZIM, and
+    either add the entry (if item is not yet inside the ZIM) or a redirect (if item with
+    same digest has already been added inside the ZIM).
+
+    This class must be configured with filters to specifiy which items paths to
+    consider. It is of course possible to consider all paths (i.e. all items) with a
+    wide regex or to operate on a subset (e.g. all images) with more precise filters.
+
+    Only the digest/path of items matching the filters are computed and stored.
+
+    The xxh32 algorithm which is known to be good at avoiding collision with minimal
+    memory and CPU footprint is used, so the sheer memory consumption will come from the
+    paths we have to keep. This hashing algorithm is not meant for security purpose
+    since one might infer original content from hashes, but this is not our use case.
+    """
+
+    def __init__(self):
+        self.filters: list[re.Pattern[str]] = []
+        self.added_items: dict[bytes, str] = {}
+
+    def add_item_or_redirect(
+        self,
+        creator: Creator,
```

Comment:
Why isn't this set on the Duplicator? Managing duplicates across multiple creators makes no sense and we have no multi-creator use case in sight anyway.

Reply:
Indeed, quite obvious, don't know how I missed this opportunity 🤣